internal/api/v1/application/portforward.go

```diff
@@ -0,0 +1,85 @@
+package application
+
+import (
+	"net/http"
+	"net/http/httputil"
+	"time"
+
+	"github.com/epinio/epinio/helpers/kubernetes"
+	"github.com/epinio/epinio/internal/application"
+	apierror "github.com/epinio/epinio/pkg/api/core/v1/errors"
+	"github.com/gin-gonic/gin"
+	thekubernetes "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+func (hc Controller) PortForward(c *gin.Context) apierror.APIErrors {
+	ctx := c.Request.Context()
+	namespace := c.Param("namespace")
+	appName := c.Param("app")
+
+	cluster, err := kubernetes.GetCluster(ctx)
+	if err != nil {
+		return apierror.InternalError(err)
+	}
+
+	if err := hc.validateNamespace(ctx, cluster, namespace); err != nil {
+		return err
+	}
+
+	app, err := application.Lookup(ctx, cluster, namespace, appName)
+	if err != nil {
+		return apierror.InternalError(err)
+	}
+
+	if app == nil {
+		return apierror.AppIsNotKnown(appName)
+	}
+
+	// app exists but has no workload to connect to
+	if app.Workload == nil {
+		return apierror.NewAPIError("Cannot connect to application without workload",
+			"", http.StatusBadRequest)
+	}
+
+	// TODO: Find podName from application and params (e.g. instance number etc).
+	// The application may have more than one pods.
```

Comment:
This should be doable now, you can already implement this. Check this for an example: https://github.com/epinio/epinio/pull/1184

Reply:
Ah yes, I forgot to ask you about this, but when I've read the TODO I thought about it! Let's try